# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1289612942 18000
#      Fri Nov 12 20:49:02 2010 -0500
# Node ID dea482126ec3bda9ab1e06003f05b4dd81ef920b
# Parent  dc5d1280ef14b24d591ed54840105883b3b59fe8
liboctave/Makefile.am: clean up some variables; don't distribute generated files

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,25 @@
+2010-11-12  John W. Eaton  <jwe@octave.org>
+
+	* Makefile.am (LIBOCTAVE_SOURCES): Delete variable.
+	(liboctave_la_SOURCES): Include $(LIBOCTAVE_CXX_SOURCES) and
+	$(LIBOCTAVE_C_SOURCES) in this list instead of
+	$(LIBOCTAVE_SOURCES).
+	(nodist_liboctave_la_SOURCES): New variable.
+	(octinclude_HEADERS): Don't include $(BUILT_INCS) in the list.
+	(noinst_octinclude_HEADERS): New variable.
+	(PRIVATE_INCS): Delete variable.
+	(INCS): Include oct-glob.h in the list.
+	(liboctave_la_SOURCES): Don't include $(PRIVATE_INCS) in the
+	list.
+	(LIBOCT_PATHSEARCH_CXX_SOURCES, LIBOCT_PATHSEARCH_C_SOURCES,
+	LIBOCT_PATHSEARCH_SOURCES): Delete variables.
+	(LIBOCTAVE_CXX_SOURCES): Include pathsearch.cc in the list.
+
 2010-11-12  John W. Eaton  <jwe@octave.org>
 
 	* Array.cc (Array<T>::cat): Simplify previous change.
 
 2010-11-11  John W. Eaton  <jwe@octave.org>
 
 	* Array.cc (Array<T>::cat): New special case for concatenating
 	empty matrices.  Bug #31615.
diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -212,16 +212,17 @@ INCS = \
   lo-utils.h \
   mach-info.h \
   oct-alloc.h \
   oct-binmap.h \
   oct-cmplx.h \
   oct-convn.h \
   oct-env.h \
   oct-fftw.h \
+  oct-glob.h \
   oct-group.h \
   oct-inttypes.h \
   oct-locbuf.h \
   oct-md5.h \
   oct-mem.h \
   oct-mutex.h \
   oct-norm.h \
   oct-openmp.h \
@@ -443,16 +444,17 @@ LIBOCTAVE_CXX_SOURCES = \
   oct-norm.cc \
   oct-passwd.cc \
   oct-rand.cc \
   oct-shlib.cc \
   oct-spparms.cc \
   oct-syscalls.cc \
   oct-time.cc \
   oct-uname.cc \
+  pathsearch.cc \
   regex-match.cc \
   sparse-sort.cc \
   sparse-util.cc \
   str-vec.cc \
   $(TI_SRC) \
   $(MATRIX_SRC)
 
 include vx-op-src.mk
@@ -479,51 +481,46 @@ LIBOCTAVE_C_SOURCES = \
   lo-cieee.c \
   lo-cutils.c \
   randgamma.c \
   randmtzig.c \
   randpoisson.c \
   tempnam.c \
   tempname.c
 
-LIBOCTAVE_SOURCES = $(LIBOCTAVE_CXX_SOURCES) $(BUILT_LIBOCTAVE_CXX_SOURCES) \
-	$(LIBOCTAVE_C_SOURCES)
-
 LIBOCT_READLINE_CXX_SOURCES = cmd-edit.cc cmd-hist.cc
 
 LIBOCT_READLINE_C_SOURCES = oct-rl-edit.c oct-rl-hist.c
 
 LIBOCT_READLINE_SOURCES = $(LIBOCT_READLINE_CXX_SOURCES) $(LIBOCT_READLINE_C_SOURCES)
 
-LIBOCT_PATHSEARCH_CXX_SOURCES = pathsearch.cc
-
-LIBOCT_PATHSEARCH_C_SOURCES = kpse-xfns.c
-
-LIBOCT_PATHSEARCH_SOURCES = pathsearch.cc
-
 LINK_DEPS = \
   $(RLD_FLAG) \
   ../libcruft/libcruft.la \
   ../libcruft/libranlib.la \
   ../libgnu/libgnu.la \
   $(SPARSE_XLIBS) \
   $(ARPACK_LIBS) \
   $(QRUPDATE_LIBS) \
   $(FFTW_XLIBS) \
   $(LAPACK_LIBS) $(BLAS_LIBS) \
   $(READLINE_LIBS) $(TERM_LIBS) \
   $(LIBGLOB) $(REGEX_LIBS) $(DL_LIBS) \
   $(FLIBS) \
   $(PTHREAD_LIBS) $(LIBS)
 
 liboctave_la_SOURCES = \
+  $(LIBOCTAVE_CXX_SOURCES) \
+  $(LIBOCTAVE_C_SOURCES) \
   $(LIBOCTAVE_SOURCES) \
   $(LIBOCT_READLINE_SOURCES) \
-  $(LIBOCT_PATHSEARCH_SOURCES) \
-  $(PRIVATE_INCS)
+  $(LIBOCT_PATHSEARCH_SOURCES)
+
+nodist_liboctave_la_SOURCES = \
+  $(BUILT_LIBOCTAVE_CXX_SOURCES)
 
 liboctave_la_LIBADD = $(LINK_DEPS)
 
 liboctave_la_CPPFLAGS = \
   @OCTAVE_DLL_DEFS@ \
   $(SPARSE_XCPPFLAGS) \
   $(FFTW_XCPPFLAGS) \
   $(ARPACK_CPPFLAGS) \
@@ -533,21 +530,23 @@ liboctave_la_LDFLAGS = -release $(versio
   -bindir $(bindir) \
   $(SPARSE_XLDFLAGS) \
   $(ARPACK_LDFLAGS) \
   $(QRUPDATE_LDFLAGS) \
   $(FFTW_XLDFLAGS)
 
 octinclude_HEADERS = \
   $(INCS) \
-  $(BUILT_INCS) \
   $(OTHER_INC) \
   $(TEMPLATE_SRC) \
   $(EXTRA_HEADERS)
 
+nodist_octinclude_HEADERS = \
+  $(BUILT_INCS)
+
 $(OPT_INC) : %.h : %.in $(top_srcdir)/mk-opts.pl
 	@echo making $@ from $<
 	@$(PERL) $(top_srcdir)/mk-opts.pl --opt-class-header $< > $@-t
 	mv $@-t $@
 
 $(VX_OP_INC) $(VX_OP_SRC) : $(srcdir)/mk-ops.awk vx-ops
 	$(AWK) -f $(srcdir)/mk-ops.awk prefix=vx $(srcdir)/vx-ops
 
diff --git a/liboctave/kpse-xfns.h b/liboctave/kpse-xfns.h
deleted file mode 100644
--- a/liboctave/kpse-xfns.h
+++ /dev/null
@@ -1,55 +0,0 @@
-/*
-
-Copyright (C) 1992, 93, 94, 95, 96, 97 Free Software Foundation, Inc.
-Copyright (C) 1993, 94, 95, 96 Karl Berry.
-Copyright (C) 1997, 1998 Free Software Foundation, Inc.
-
-This library is free software; you can redistribute it and/or
-modify it under the terms of the GNU Library General Public
-License as published by the Free Software Foundation; either
-version 2 of the License, or (at your option) any later version. 
-
-This library is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-Library General Public License for more details.
-
-You should have received a copy of the GNU Library General Public
-License along with this library; if not, write to the Free Software
-Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
-02110-1301, USA.  */
-
-#if !defined (octave_kpse_xfns_h)
-#define octave_kpse_xfns_h 1
-
-/* Define the characters which separate components of
-   filenames and environment variable paths.  */
-
-/* What separates filename components?  */
-#ifndef DIR_SEP
-#ifdef DOSISH
-/* Either \'s or 's work.  Wayne Sullivan's web2pc prefers /, so we'll
-   go with that.  */
-#define DIR_SEP '/'
-#define DIR_SEP_STRING "/"
-#define IS_DEVICE_SEP(ch) ((ch) == ':')
-#define NAME_BEGINS_WITH_DEVICE(name) ((name.length()>0) && IS_DEVICE_SEP((name)[1]))
-/* On DOS, it's good to allow both \ and / between directories.  */
-#define IS_DIR_SEP(ch) ((ch) == '/' || (ch) == '\\')
-#else
-#define DIR_SEP '/'
-#define DIR_SEP_STRING "/"
-#endif /* not DOSISH */
-#endif /* not DIR_SEP */
-
-#ifndef IS_DIR_SEP
-#define IS_DIR_SEP(ch) ((ch) == DIR_SEP)
-#endif
-#ifndef IS_DEVICE_SEP /* No `devices' on, e.g., Unix.  */
-#define IS_DEVICE_SEP(ch) 0 
-#endif
-#ifndef NAME_BEGINS_WITH_DEVICE
-#define NAME_BEGINS_WITH_DEVICE(name) 0 
-#endif
-
-#endif
